resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java

```diff
@@ -209,28 +211,9 @@ public RestClientBuilder executorService(ExecutorService executor) {
                     envProxyHost,
                     Integer.parseInt(System.getProperty("http.proxyPort", "80")));
         } else {
-            // Search for proxy settings passed in the client builder, if passed and use them if found
-            String userProxyHost = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_HOST))
-                    .filter(String.class::isInstance)
-                    .map(String.class::cast)
-                    .orElse(null);
-
-            Integer userProxyPort = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_PORT))
-                    .filter(Integer.class::isInstance)
-                    .map(Integer.class::cast)
-                    .orElse(null);
-
-            String userProxyScheme = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_SCHEME))
-                    .filter(String.class::isInstance)
-                    .map(String.class::cast)
-                    .orElse(null);
-
-            if (userProxyHost != null && userProxyPort != null) {
-                resteasyClientBuilder = builderDelegate.defaultProxy(userProxyHost, userProxyPort, userProxyScheme);
-            } else {
-                //no proxy
-                resteasyClientBuilder = builderDelegate;
-            }
+            selectHttpProxy().ifPresent(proxyAddress -> builderDelegate.defaultProxy(
+                    proxyAddress.getHostString(), proxyAddress.getPort()));
```

Comment:
Is this really meant to replace the check for user provided proxy properties too? Basically wondering about the difference between this PR and #2332 .